### PR TITLE
Update docs on ChangeNotifier.dispose and KeepAliveHandle.release

### PR DESCRIPTION
**Pull request by dnfield**
See https://github.com/flutter/flutter/pull/108227/files#r930193822

@polina-c FYI

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/automatic_keep_alive.dart [hunk 1]
```diff
@@ -321,6 +321,9 @@ class KeepAliveNotification extends Notification {
 class KeepAliveHandle extends ChangeNotifier {
   /// Trigger the listeners to indicate that the widget
   /// no longer needs to be kept alive.
+  ///
+  /// This method does not call [dispose]. When the handle is not needed
+  /// anymore, it must be [dispose]d regardless of whether notifying listeners.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/automatic_keep_alive.dart:326]: Should we just fix this by deprecating release and overriding `dispose` to call `notifyListeners` before `super.dispose`?

**Review by goderbauer — COMMENTED** [review]
LGTM since documenting the current behavior is not making it worse. I think we can do better by improving the actual API, though.

File: packages/flutter/lib/src/widgets/automatic_keep_alive.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter/lib/src/widgets/automatic_keep_alive.dart:326]: I think that makes sense. I wasn't clear on what if anything that would break.

Comment (dnfield) [reply packages/flutter/lib/src/widgets/automatic_keep_alive.dart:326]: Do you want me to do that in this PR or should we file an issue/follow up?

Comment (goderbauer) [reply packages/flutter/lib/src/widgets/automatic_keep_alive.dart:326]: I would prefer doing it in this PR, but I would be fine either way.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
ARgh, we should not be calling both release and dispose now though since that will notify listeners twice...

File: packages/flutter/lib/src/widgets/automatic_keep_alive.dart [hunk 2]
```diff
@@ -354,7 +367,6 @@ mixin AutomaticKeepAliveClientMixin<T extends StatefulWidget> on State<T> {
 
   void _releaseKeepAlive() {
     // Dispose and release do not imply each other.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/automatic_keep_alive.dart:369]: well this comment definitely seems out of place now

**Pull request merged**